core/src/engine/task/work_stealing_queue/global_queue.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <cstddef>
+
+#include <moodycamel/blockingconcurrentqueue.h>
+#include <moodycamel/lightweightsemaphore.h>
+
+#include <userver/utils/span.hpp>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace engine {
+
+template <typename T>
+class GlobalQueue final {
+ public:
+  GlobalQueue() = default;
+
+  std::size_t GetSize() const noexcept { return size_.load(); }
+
+  void Push(T* const ctx) {
```

Comment:
Let's either:
1. make it not templated, use forward-declared `TaskContext`
2. leave it templated, but `T` will stand for `TaskContext*` instead of `TaskContext`, it will be more general this way